test/subdataframe.jl

```diff
@@ -293,8 +293,9 @@ end
     @test_throws ArgumentError SubDataFrame(sdf, true, :)
     @test_throws ArgumentError SubDataFrame(sdf, Integer[true], 1)
 
-    # this is an error in Julia Base
-    SubDataFrame(sdf, Integer[true, true, true], :) == SubDataFrame(sdf, [1, 1, 1], :)
+    if VERSION > v"1.6"
+        @test_throws ArgumentError SubDataFrame(sdf, Integer[true, true, true], :)
+    end
```

Comment:
And what happens before 1.6? :-)

Reply:
Before (and including 1.6) 1.7 it works - see the old test that got removed.

Reply:
I mean, shouldn't this be tested too?

Reply:
OK - I have reverted it (although it was kept mainly to keep track of the bug in Julia Base).

A follow up PR in Julia Base that should be fixed is https://github.com/JuliaLang/julia/pull/39962 (I am pinging you as it seem no one commented on it).